pkg/divert/driver.go

```diff
@@ -36,19 +37,25 @@ type Driver interface {
 	UpdateVirtualService(vs *istioNetworkingV1beta1.VirtualService)
 }
 
-func New(m *model.Manifest, c kubernetes.Interface) (Driver, error) {
+func New(divert *model.DivertDeploy, name, namespace string, c kubernetes.Interface) (Driver, error) {
 	if !okteto.IsOkteto() {
 		return nil, oktetoErrors.ErrDivertNotSupported
 	}
 
-	if m.Deploy.Divert.Driver == constants.OktetoDivertWeaverDriver {
-		return weaver.New(m, c), nil
+	if divert.Driver == constants.OktetoDivertWeaverDriver {
+		return weaver.New(divert, name, namespace, c), nil
 	}
 
 	ic, err := virtualservices.GetIstioClient()
 	if err != nil {
 		return nil, fmt.Errorf("error creating istio client: %w", err)
 	}
 
-	return istio.New(m, c, ic), nil
+	return istio.New(divert, name, namespace, c, ic), nil
+}
+
+// NewNoop returns a new noop driver for divert. Useful for cases in which divert is not defined
+// in the manifest but the flow needs a driver
+func NewNoop() Driver {
```

Comment:
Created this to simplify the logic in the compose translation and prevent the `if divertDriver != nil`